### PR TITLE
Add minitest backtrace filter

### PR DESCRIPTION
**Pull request by andyw8**
Closes https://github.com/Shopify/spoom/issues/456

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -348,6 +348,16 @@ Find all the symbols for a file:
 puts client.document_symbols("file://path/to/my/file.rb")
 ```
 
+### Backtrace Filtering
+
+Spoom provides a backtrace filter for Minitest to remove the Sorbet frames from test failures, giving a more readable output. To enable it:
+
+```ruby
+# test/test_helper.rb
+require "spoom/minitest/backtrace_filter"
```
Comment (andyw8) [inline README.md:357]: (wondering where this should be a manual require or not)

File: lib/spoom/minitest/backtrace_filter.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "minitest"
+
+module Spoom
+  module BacktraceFilter
+    class Minitest < ::Minitest::BacktraceFilter
```
Comment (andyw8) [inline lib/spoom/minitest/backtrace_filter.rb:8]: I originally had `Spoom::Minitest::BacktraceFilter` but that causes typecheck problems with tests in the `Spoom` namespace that inherit from `Minitest::Test`.

File: lib/spoom.rb [hunk 3]
```diff
@@ -18,4 +18,5 @@ class Error < StandardError; end
 require "spoom/deadcode"
 require "spoom/sorbet"
 require "spoom/cli"
+require "spoom/minitest/backtrace_filter"
```
Comment (vinistock) [inline lib/spoom.rb:21]: I'd make this require manual and not included in the default require.

File: lib/spoom/minitest/backtrace_filter.rb (hunk 2 above)
Comment (andyw8) [reply lib/spoom/minitest/backtrace_filter.rb]: I think this way is ok, have renamed the file to match.

**Review by vinistock — APPROVED** [review]
I think this is fine, but I'll let Alex decide if he prefers the default require

**Review by Morriar — APPROVED** [review]

**Pull request merged**